_specs/ecip-1084.md

```diff
@@ -0,0 +1,139 @@
+root issue which is the malincentive of the miner vote.
+
+A protocol change to set a hard-cap would resolve the unsustainability issue for
+ETC and bring the protocol into alignment with ETC philosophy, rejecting this bad
+inheritance from ETH and back in line with Bitcoin's sustainable approach.
+Unlike BTC however, we would define a long-term growth curve to reflect the
+constantly improving hardware - even with the same protocol and same client
+implementations.
+
+
+## Specification
+
+At block number ACTIVATION_BLOCK on ETC mainnet, switch the gas-limit calculation
+to be one of the following two options:
+
+1. gaslimit = INITIAL_GASLIMIT + (block_number - ACTIVATION_BLOCK) * GASLIMIT_CURVE_RATE
+2. gaslimit = INITIAL_GASLIMIT + (block_number - ACTIVATION_BLOCK) * 2 ^ GASLIMIT_CURVE_RATE
+
+Let ACTIVATION_BLOCK = TBD
+Let INITIAL_GASLIMIT = 2,000,000 (subject to change)
+Let GASLIMIT_CURVE_RATE = TBD
+
+
+## Backwards compatibility
+
+There is certainly a concern that some smart contracts which used to work will cease
+to work (until some future point where the raising gaslimit makes them viable again)
+and we need to quantify this risk.
+
+There is a concern that it will not be possible to migrate ETH smart contracts to ETC
+because of the lower gaslimit.  Straw polls on Twitter seem to indicated that 1M or 2M
+is adequate for most smart contracts out there for their heaviest transactions with the
+sole exception of transactions for contract deployment, which can hit 4M or more.  In
+that case we have no answer yet other than spliting contracts up into multiple libraries.
+
+It looks like some kind of "Webpack for Solidity" would be workable, where we could add
+opcodes to deploy chunks of bytecode across several transactions, with new opcodes for
+START_DEPLOY, DEPLOY_CHUNK and END_DEPLOY or similar which could be used transparency
+internal to the development framework (Truffle, Embark, Brownie, etc) and
+compiler (SOLC, SOLL, Vyper, etc) to generate different bytecode depending on the
+gaslimit of the target chain.   This is just an idea for the time being and needs its
+own new ECIP too.
+
+Aragon is probably the perfect guinea pig for addressing these concerns because they have
+large smart contracts and have just recently been the victim of backwards-compatibility
+breaks in ETH as part of the Istanbul HF.  If we can provide a great story for ETH to ETC
+migration with well engineered changes spanning frameworks, compiler and runtime then
+that will be a delightful win-win for all parties.  As Jorge said, while breaking smart
+contracts up is the right thing to do, it should not be REQUIRED.  To require changes
+to existing script is a break in the implicit social contract:
+
+https://www.coindesk.com/ethereums-istanbul-upgrade-will-break-680-smart-contracts-on-aragon
+
+
+### Implementation
+
+There is no implementation of this proposal yet because we are missing the specific details
```

Comment:
Is this draft waiting for these specifics?

Reply:
No - not waiting on anything.
Details to be fleshed out in public within this Draft.
Looking to gather consensus around the concept.